tests/test_optimize.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+from unittest import TestCase
+
+from aspire.utils import ensure
+from aspire.utils.matlab_compat import m_reshape
+from aspire.utils.optimize import conj_grad
+
+
+def identity(x):
+    return x
+
+
+def precond_fun(S, x):
+    p = np.size(S, 0)
+    ensure(np.size(x) == p*p, 'The sizes of S and x are not consistent.')
+    x = m_reshape(x, (p, p))
+    y = S @ x @ S
+    y = m_reshape(y, (p ** 2,))
+    return y
+
+
+def apply(A, x):
+    p = np.size(A[0], 0)
+    x = m_reshape(x, (p, p))
+    y = np.zeros_like(x)
+    for k in range(0, len(A)):
+            y = y + A[k] @ x @ A[k].T
+    y = m_reshape(y, (p ** 2,))
+    return y
+
+
+class OptimizeTestCase(TestCase):
+    def setUp(self):
+
+        self.cg_opt = {
+            'shrinker': 'frobenius_norm', 'verbose': 0, 'max_iter': 250,
+            'iter_callback': [], 'store_iterates': False,
+            'rel_tolerance': 1e-12, 'precision': 'float64',
+            'preconditioner': identity
+        }
+
+        self.cg_opt["preconditioner"] = lambda x: precond_fun(self.S, x)
+
+        self.A = [
```

Comment:
Generate these arrays and matrices using random numbers. No need to hardcode them.

Reply:
Done in new commit 6f43523.

Reply:
What about `b`, `x`, and `x_comp`?

Reply:
All matrices (A and A_comp) and vectors (x and x_comp) were generated from random numbers in commit 6d8aeeb.  b and b_comp were evaluated from apply(A, x) and served as inputs to estimate x and x_comp.

Reply:
We just talked .. @junchaoxia will make A/b/b_comp completely random (with no fixed seed) with the correct dtype/shape, and only check the solution. Also the `apply` function will go away.

Reply:
In new commit 4947d02, all A and b are generated in different values for different unit tests.